### DISCUSSION TITLE
How to manage and compare AbdulRazzakGhan parameterisation and BetancourtNenes

### DISCUSSION THREAD

**Discussion by oyvindseland** [NorESMhub/CAM#171] [General]
I have at last gone back to the activation code and think (hope?) that I understand the assumptions and structure now.

Why is it so difficult to compare and combine the two different parameterisations:
The activation parameterisation has two steps to it. It calculates the fraction that is activated, but in order to get a tendency (number / second) it need to use the upwards moment to calculate the time tendency. The flux of aerosols is defined as a vertical velocity.
The reason why I do think I understand the structure now is that I went through the units of the equations and  was able to get from the two output values (fraction (so unitless) and m/s, to (kg of air)-1 * seconds -1. The main trick is that part of the calculations deal with fractional column values so it does not have to bother with actual concentrations.
This vertical velocity should be / is assumed be the same vertical velocity that drives the supersaturation. There is an exception for the lowest model layer, which has it own parameterisation which according to the comments none really understands.

For ARG parameterisation both of these quantities are calculated in  activate_modal.
The problem is that the two quantities are not treated independently and this was what confused me to the point that BM was implemented outside activate_modal despite needing the same input / output fields.
 The higher vertical velocities in the pdf is assumed to be less efficient in driving the supersaturation. Since the vertical flux used for calculating the flux is supposed to follow what drives the supersaturation the "vertical velocity flux" is reduced from the integrated value to this modified value. If there is no pdf then the flux = vertical velocity. In other words it looks like the simplification I did in the first place was valid. 
BN does not use this assumption so to take into account the distribution it is enough to integrate the speed over the pdf so a good portion of  fraction of the ARG calculations can be bypassed and replaced by the pdf integral (i.e erf function).
Also making it simpler, this particular assumption can be seen as inherent to the ARG parameterision so should not be included in BN.

For coding I suggest that BN should be moved into the same type of subroutine as the present activate_modal subroutine, but not not inside the same routine which I think would be hard to read.

@ingvisau @gold2718 @MichaelSchulzMETNO

**Comment by ingvisau** [comment]
Thank you so much for the effort to clear this up! I think I understand, at least partly. I have already moved around on the BN code so that it can be called by an activate_modal_BN subroutine. I just haven't been able to untangle all of the code in activate_modal(_ARG). 
But if I understand you correctly, we would have comparable runs if we use the flux parameterization for BN, just with the integral of the vertical velocity distribution, and move it to activate_modal_BN?

**Reply by oyvindseland** [reply]
Yes, I think that should cover it.
For the first round of code you can even keep the flux=wbar as a placeholder also for the pdf version, just do not forget about it.

**Reply by ingvisau** [reply]
But are you saying I can also do a similar parameterization for ARG? Because I don't know what part of the code in activate_modal_ARG I can replace with that... It would be delightful if I could get some help looking at the code in the original activate_modal (I have been staring myself blind on it), and maybe remove some parts/loops we don't need for this :)

Btw, if we figure that part out, I agree that we could look at the limit on the higher vertical velocities as an inherent ARG thing :+1:

**Comment by oyvindseland** [comment]
My suggestion out of this:
-Check if defaults, BN and ARG use single updraft or pdf.
If default BN use single updraft then existing BN results/ simulations can be used.
If default ARG use single updraft then also the existing ARG simulations can be used directly
If default ARG use pdf updraft then the existing results can be used but one should consider creating experiments with single updraft ARG to understand differences.

If default BN use pdf updraft than the effective updraft should be found by integrating the distribution.(Not quite trivial but equations do exist)

**Comment by oyvindseland** [comment]
ARG is single updraft
